lib/apress/selenium_eti/spec/company_site/eti/product_price_spec.rb

```diff
@@ -39,88 +42,122 @@
-          elsif @cs_eti_page.project_blizko?
-            expect(@cs_eti_page.product_price(@name)).to eq 'до ' + @price_to[:to].to_s + ' ₽'
+          if @cs_eti_table.project_pulscen?
+            expect(@cs_eti_table_products.price(@product)).to eq("до #{price[:price_max]} руб.")
+          elsif @cs_eti_table.project_blizko?
+            expect(@cs_eti_table_products.price(@product)).to eq("до #{price[:price_max]} ₽")
           else
             raise ArgumentError, 'locator not found'
           end
         end
       end
 
       context 'когда заполняем "от" и "до"' do
-        before do
-          @price_from_to = {from: Faker::Number.number(digits: 2), to: Faker::Number.number(digits: 3)}
-          @cs_eti_page.set_price_from_to(@price_from_to)
-          @cs_eti_page.wait_saving
-          @cs_eti_page.search_product(@name)
+        let(:price) do
+          {
+            type: :range,
+            price: Faker::Number.number(digits: 2),
+            price_max: Faker::Number.number(digits: 3)
+          }
         end
 
         it 'введеная цена отображается' do
-          expect(@cs_eti_page.price_value).to include @price_from_to[:from].to_s, @price_from_to[:to].to_s
+          expect(@cs_eti_table_products.price(@product)).to include(price[:price].to_s, price[:price_max].to_s)
         end
       end
     end
 
     context 'когда цена со скидкой', story: 'когда цена со скидкой' do
-      before do
-        @discount_price = {previous: Faker::Number.number(digits: 3), discount: Faker::Number.number(digits: 2)}
-        @cs_eti_page.set_discount_price(@discount_price)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.search_product(@name)
+      let(:price) do
+        {
+          type: :discount,
+          price: Faker::Number.number(digits: 3),
+          new_price: Faker::Number.number(digits: 2),
+          expires_at: Time.now.strftime('%d.%m.%Y')
+        }
       end
 
       it 'введеная цена отображается' do
-        expect(@cs_eti_page.discount_price_value).to include @discount_price[:discount].to_s
-        expect(@cs_eti_page.previous_price_value).to include @discount_price[:previous].to_s
+        expect(@cs_eti_table_products.price(@product)).to include(price[:price].to_s, price[:new_price].to_s)
+      end
+    end
+  end
+
+  describe 'Установка оптовых цен', feature: 'company_site/eti/product_price_spec: Установка оптовых цен' do
+    before do
+      @name = Faker::Number.number(digits: 5).to_s
+      @cs_eti_table_products.add_product(name: @name, wholesale_price: wholesale_price)
+      @cs_eti_header.search_product(@name, exact: true)
+      @product = @cs_eti_table_products.product(name: @name)
+    end
+
+    context 'когда точная оптовая цена', story: 'когда точная оптовая цена' do
+      let(:wholesale_price) do
+        {
+          price: Faker::Number.number(digits: 2),
+          min_qty: Faker::Number.non_zero_digit
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.